maestro-ios/src/main/java/ios/xctest/XCTestIOSDevice.kt

```diff
@@ -218,95 +182,57 @@ class XCTestIOSDevice(
+        if (mutable.containsKey("all")) {
+            val value = mutable.remove("all")
+            allPermissions.forEach {
+                when (value) {
+                    "allow" -> mutable.putIfAbsent(it, "allow")
+                    "deny" -> mutable.putIfAbsent(it, "deny")
+                    "unset" -> mutable.putIfAbsent(it, "unset")
+                    else -> throw IllegalArgumentException("Permission 'all' can be set to 'allow', 'deny' or 'unset', not '$value'")
                 }
             }
-
-            client.setPermissions(mutable)
         }
+
+        execute { client.setPermissions(mutable) }
     }
 
     override fun eraseText(charactersToErase: Int) {
         val appIds = getInstalledApps()
-        client.eraseText(charactersToErase, appIds).use {}
+        execute { client.eraseText(charactersToErase, appIds) }
     }
 
-    private fun activeAppId(): String? {
-        val appIds = getInstalledApps()
-        logger.info("installed apps: $appIds")
-
-        return client.runningAppId(appIds).use { response ->
-            response.body.use { body ->
-                val runningAppBundleId = if (response.isSuccessful) {
-                    body?.let {
-                        val responseBody: GetRunningAppIdResponse = mapper.readValue(
-                            String(it.bytes()),
-                            GetRunningAppIdResponse::class.java
-                        )
-                        val runningAppId = responseBody.runningAppBundleId
-                        logger.info("Running app id response received $runningAppId")
-                        runningAppId
-                    }
-                } else {
-                    val bodyString = response.body?.let { String(it.bytes()) } ?: ""
-                    val code = response.code
-                    logger.info("request to resolve running app id failed with exception - Code: $code Body: $bodyString")
-
-                    return null
-                }
-
-                logger.info("found running app id $runningAppBundleId")
+    private fun activeAppId(): String {
+        return execute {
+            val appIds = getInstalledApps()
+            logger.info("installed apps: $appIds")
 
-                runningAppBundleId
-            }
+            client.runningAppId(appIds).runningAppBundleId
         }
     }
 
-    class IllegalArgumentSnapshotFailure : Throwable("Failed to capture view hierarchy due to kAXErrorIllegalArgument")
-    class InputFieldNotFound : Throwable("Unable to find focused input field")
-    class UnknownFailure(errorResponse: String) : Throwable(errorResponse)
+    private fun <T> execute(call: () -> T): T {
+        return try {
+            call()
+        } catch (appCrashException: XCUITestServerError.AppCrash) {
+            throw IOSDeviceErrors.AppCrash(
+                "App crashed, please check diagnostic logs: " +
+                        "/Library/Logs/DiagnosticReports directory"
```

Comment:
@artem888 this is a message which users will see on CLI in case their app crash in between flow. Any suggestions what we can show here or this looks good?

Reply:
Fixed